CMakeLists.txt

```diff
@@ -45,6 +45,7 @@ elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}
     if (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 5.0)
         message(FATAL_ERROR "Clang version must be at least 5.0!")
     endif()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wsign-compare -Wrange-loop-analysis")
```

Comment:
did you mean to add `-Wall`? (wasn't in PR desc)

Reply:
After cleaning up these warnings, what would be the thoughts on turning any new warnings into errors `-Werror`?

Reply:
`-Werror` by default is very user hostile, I'd hope it'd be an option defaulted to off

Reply:
(and potentially enabled in CI)